internal/cmdliveinit/cmdliveinit.go

```diff
@@ -182,54 +187,70 @@ func (c *ConfigureInventoryInfo) Run(ctx context.Context) error {
 	if !isEmpty && !force {
+		return errors.E(op, p.UniquePath, &InvInKfExistsError{})
+	}
+	// Set the Kptfile inventory to be nil since we force write to resourcegroup instead.
+	kf.Inventory = nil
+
+	// Validate the inventory values don't already exist in Resourcegroup.
+	if rg != nil && !force {
 		return errors.E(op, p.UniquePath, &InvExistsError{})
 	}
-	// Finally, set the inventory parameters in the Kptfile and write it.
-	kf.Inventory = inv
-	if err := kptfileutil.WriteFile(p.UniquePath.String(), kf); err != nil {
+	// Initialize new resourcegroup object, as rg should have been nil.
+	rg = &rgfilev1alpha1.ResourceGroup{ResourceMeta: rgfilev1alpha1.DefaultMeta}
+	// // Finally, set the inventory parameters in the ResourceGroup object and write it.
+	rg.Name = inv.Name
+	rg.Namespace = inv.Namespace
+	if inv.InventoryID != "" {
+		rg.Labels = map[string]string{rgfilev1alpha1.RGInventoryIDLabel: inv.InventoryID}
+	}
+	if err := writeRGFile(p.UniquePath.String(), rg, filename); err != nil {
 		return errors.E(op, p.UniquePath, err)
 	}
+
+	// Rewrite Kptfile if inventory exists
+	if !isEmpty {
+		if err := kptfileutil.WriteFile(p.UniquePath.String(), kf); err != nil {
+			return errors.E(op, p.UniquePath, err)
+		}
+	}
+
 	return nil
 }
 
-// generateID returns the string which is a SHA1 hash of the passed namespace
-// and name, with the unix timestamp string concatenated. Returns an error
-// if either the namespace or name are empty.
-func generateID(namespace string, name string, t time.Time) (string, error) {
-	const op errors.Op = "cmdliveinit.generateID"
-	hashStr, err := generateHash(namespace, name)
+func writeRGFile(dir string, rg *rgfilev1alpha1.ResourceGroup, filename string) error {
+	const op errors.Op = "cmdliveinit.writeRGFile"
+	b, err := yaml.MarshalWithOptions(rg, &yaml.EncoderOptions{SeqIndent: yaml.WideSequenceStyle})
 	if err != nil {
-		return "", errors.E(op, err)
+		return err
 	}
-	timeStr := strconv.FormatInt(t.UTC().UnixNano(), 10)
-	return fmt.Sprintf("%s-%s", hashStr, timeStr), nil
-}
-
-// generateHash returns the SHA1 hash of the concatenated "namespace:name" string,
-// or an error if either namespace or name is empty.
-func generateHash(namespace string, name string) (string, error) {
-	const op errors.Op = "cmdliveinit.generateHash"
-	if len(namespace) == 0 || len(name) == 0 {
-		return "", errors.E(op,
-			fmt.Errorf("can not generate hash with empty namespace or name"))
+	if _, err := os.Stat(filepath.Join(dir, filename)); err != nil && !goerrors.Is(err, os.ErrNotExist) {
+		return errors.E(op, errors.IO, types.UniquePath(dir), err)
 	}
-	str := fmt.Sprintf("%s:%s", namespace, name)
-	h := sha1.New()
-	if _, err := h.Write([]byte(str)); err != nil {
-		return "", errors.E(op, err)
+
+	// fyi: perm is ignored if the file already exists
+	err = ioutil.WriteFile(filepath.Join(dir, filename), b, 0600)
+	if err != nil {
+		return errors.E(op, errors.IO, types.UniquePath(dir), err)
 	}
```

Comment:
If file already exist, we shall warn user, it might have been used for difference purpose.

Reply:
The gating logic for the resourcegroup file already exists occurs much earlier. When `live init` is run, it determines if the resourcegroup.yaml file already exists. If it does, and contains an inventory, then an error is thrown unless the `--force` flag is enabled. If the file does not contain an inventory, then it will throw an error that the file could not be parsed as a resourcegroup inventory object.